doc: some minor touch-ups (nitpicks) in the contributor docs (#41966)
nodejs/node: doc/contributing/maintaining-types-for-nodejs.md

# Maintaining types for Node.js

While JavaScript is an untyped language, there are a number of complementary

> technologies like [TypeScript][] and
> [Flow][] which allow developers to use types within their

Reviewer: ```suggestion
like [TypeScript][] and [Flow][], which allow developers to annotate the source
code of their JavaScript projects. While many people don't annotate their code,
```

> JavaScript projects. While many people don't use types, there are enough
> who do that the project has agreed it's important to work towards having

Reviewer: ```suggestion
or make use of annotations at all, there are enough who do that the project has
agreed it's important to work towards having [suitable types for end-users][].
```

> [suitable types for end-users][].
> 

Reviewer: ```suggestion

```

maintained.

> The different options were discussed as part of the

Reviewer: ```suggestion
The different options were discussed as part of the [next-10][] effort and it
was agreed that maintaining them externally is the best approach.
```

> [next-10][]
> effort and it was agreed that maintaining them externally is the best approach.

Reviewer: ```suggestion

```

There is an ongoing effort to add additional markdown constraints and
then update the flow in order to be able to generate a better
JSON output.

[doc/api]: https://github.com/nodejs/node/tree/master/doc/api
[documentation-style-guide]: https://github.com/nodejs/node/blob/master/doc/README.md#documentation-style-guide
[tools/doc]: https://github.com/nodejs/node/tree/master/tools/doc
[nodejs.org]: https://nodejs.org/en/docs/
[next-10]: https://github.com/nodejs/next-10/blob/main/meetings/summit-nov-2021.md#suitable-types-for-end-users

> [suitable types for end users]: https://github.com/nodejs/node/blob/master/doc/contributing/technical-priorities.md#suitable-types-for-end-users

Reviewer 2: ```suggestion
[suitable types for end-users]: https://github.com/nodejs/node/blob/master/doc/contributing/technical-priorities.md#suitable-types-for-end-users
```

> [doc/api]: https://github.com/nodejs/node/tree/master/doc/api
> [documentation-style-guide]: https://github.com/nodejs/node/blob/master/doc/README.md#documentation-style-guide
> [tools/doc]: https://github.com/nodejs/node/tree/master/tools/doc
> [nodejs.org]: https://nodejs.org/en/docs/
> [next-10]: https://github.com/nodejs/next-10/blob/main/meetings/summit-nov-2021.md#suitable-types-for-end-users
> [suitable types for end users]: https://github.com/nodejs/node/blob/master/doc/contributing/technical-priorities.md#suitable-types-for-end-users
> [Flow]: https://flow.org/
> [TypeScript]: https://www.typescriptlang.org/

Reviewer 2: Re-order in ASCII order, and use `HEAD` instead of `master` (in case the default branch name ever changes).
```suggestion
[Flow]: https://flow.org/
[TypeScript]: https://www.typescriptlang.org/
[doc/api]: https://github.com/nodejs/node/tree/HEAD/doc/api
[documentation-style-guide]: https://github.com/nodejs/node/blob/HEAD/doc/README.md#documentation-style-guide
[next-10]: https://github.com/nodejs/next-10/blob/HEAD/meetings/summit-nov-2021.md#suitable-types-for-end-users
[nodejs.org]: https://nodejs.org/en/docs/
[suitable types for end-users]: https://github.com/nodejs/node/blob/HEAD/doc/contributing/technical-priorities.md#suitable-types-for-end-users
[tools/doc]: https://github.com/nodejs/node/tree/HEAD/tools/doc
```
